ocs_ci/ocs/must_gather/must_gather.py

```diff
@@ -0,0 +1,171 @@
+import os
+import logging
+import shutil
+import tempfile
+import re
+from pathlib import Path
+
+from ocs_ci.ocs.resources.pod import get_all_pods
+from ocs_ci.ocs.utils import collect_ocs_logs
+from ocs_ci.ocs.must_gather.const_must_gather import GATHER_COMMANDS_VERSION
+from ocs_ci.ocs.ocp import get_ocs_parsed_version
+
+
+logger = logging.getLogger(__name__)
+
+
+class MustGather(object):
+    """
+    MustGather Class
+
+    """
+    def __init__(self):
+        self.type_log = None
+        self.root = None
+        self.files_path = dict()
+        self.empty_files = list()
+        self.files_not_exist = list()
+        self.files_content_issue = list()
+
+    @property
+    def log_type(self):
+        return self.type_log
+
+    @log_type.setter
+    def log_type(self, type_log):
+        if not isinstance(type_log, str):
+            raise ValueError('log type arg must be a string')
+        self.type_log = type_log
+
+    def collect_must_gather(self):
+        """
+        Collect ocs_must_gather and copy the logs to a temporary folder.
+
+        """
+        temp_folder = tempfile.mkdtemp()
+        collect_ocs_logs(dir_name=temp_folder, ocp=False)
+        self.root = temp_folder + '_ocs_logs'
+
+    def search_file_path(self):
+        """
+        Search File Path
+
+        """
+        files = GATHER_COMMANDS_VERSION[(self.type_log, get_ocs_parsed_version())]
```

Comment:
so this one will fail for OCS 4.7, right? How we will manage new OCS versions?
Also I see you added data only for OCS 4.6 and 4.5, but we have to support OCS n-2, n-1, n, + QE is testing also future release n+1 so mentioned 4.7. 

So I guess that the test should be skipped for the version which are not implemented. As I don't expect we would like to invest much in 4.4 now we can skip TC for OCS < 4.5. But what about mentioned 4.7 and any other future release?

Reply:
@petr-balogh 
*I added OCS 4.7 to `GATHER_COMMANDS_VERSION` dictionary.

The Procedure to add new version:
1.New Version created
2.Add the new version to `GATHER_COMMANDS_VERSION` dictionary
3.Ask the developer about new files on new version
4.Add file names to relevant list

In this way, we need to change only `ocs_ci/ocs/must_gather/const_must_gather.py` file

What do you think?

Reply:
I am for to put this info to the docstring of this file ;)
So everyone will know what to do when new version will come.

Also like to see in documentation links where we can check the commands like:
https://github.com/openshift/ocs-operator/blob/release-4.6/must-gather/collection-scripts/gather_ceph_resources#L35-L90

And if there are more of such links, sources I would like to see it in docstring, mahybe in const_must_gather.py